### ISSUE TITLE
if-then-else that doesn't fit onto one line inserts 5 spaces

### ISSUE THREAD

**Issue by nicolasg-da** [commercialhaskell/hindent#450]
# What happens

if-then-else that doesn't fit onto one line inserts 5 spaces

```
module Issue where

instaltheFunction arg =
    if cond
        then pure theResult
        else liftIO $
             LongModuleName.andThisIsAQuiteLongFunction $ \lArg ->
                 someStuffThatDoesntFitOntoOneLineWithTheLongFunction
                     ohAParam
                     lArg
```
It's easier to see in this screenshot:
<img width="674" alt="screen shot 2017-09-27 at 09 50 24" src="https://user-images.githubusercontent.com/20789770/30902260-e5f008ca-a36a-11e7-9c22-9dfa615d4836.png">

# What i would expect

Something more like this:
```
module Issue where

instaltheFunction arg =
    if cond
    then pure theResult
    else
        liftIO $ LongModuleName.andThisIsAQuiteLongFunction $ \lArg ->
            someStuffThatDoesntFitOntoOneLineWithTheLongFunction
                ohAParam
                lArg
```

Putting the `then` and `else` on the same level as the `if` also saves some horizontal space.

**Comment by sighingnow** [comment]
I don't thinks it's reasonable to align `if` and `then`/`else` with the same column. This layout would hidden the nested structure of code blocks and not eligible for readers.

**Comment by tfausak** [comment]
Yeah, I think `then`/`else` needs to be indented, but the problem here is that the indentation *inside* the `then`/`else` is context sensitive. Everything after the `else` should be bumped down to the next line and indented with the specified `--indent-size`.

``` hs
if p
    then t
    else
        some long expression only indented by 4 spaces rather than 5
```

**Comment by sighingnow** [comment]
Yes, as @tfausak commented above, it indeed make me feel uncomfortable. I'll try to push a patch for this issue.
